tools/optimizer/optimizer.cpp

```diff
@@ -3304,40 +3306,61 @@ void registerizeHarder(Ref ast) {
     };
     std::unordered_map<IString, JuncVar> junctionVariables;
 
-    auto initializeJunctionVariable = [&](IString name) {
-      junctionVariables[name].conf.reserve(asmData.locals.size());
+    auto getJunctionVariable = [&](IString name) -> JuncVar& {
+      bool created = junctionVariables.count(name) == 0;
+      JuncVar& jvar = junctionVariables[name];
+      if (created) jvar.conf.reserve(asmData.locals.size());
+      return jvar;
     };
 
     for (size_t i = 0; i < junctions.size(); i++) {
       Junction& junc = junctions[i];
+
+      // Pre-compute the possible conflicts and links for each block rather
+      // than checking potentially impossible options for each var
+      std::unordered_map<IString, std::vector<Block*>> possibleBlockConflicts;
+      std::unordered_map<IString, std::vector<Block*>> possibleBlockLinks;
+      for (auto b : junc.outblocks) {
+        Block* block = blocks[b];
+        Junction& jSucc = junctions[block->exit];
+        for (auto otherName : jSucc.live) {
```

Comment:
style nit: ISTM this could be just "name" rather than "otherName" now that it's been lifted out into a different scope